src/devices/bus/mc10/mcx128.cpp

```diff
@@ -108,26 +103,30 @@ class mc10_pak_mcx128_device :
 
 };
 
-DEFINE_DEVICE_TYPE_PRIVATE(MC10_PAK_MCX128, device_mc10cart_interface, mc10_pak_mcx128_device, "mc10_mcx128", "Darren Atkinson's MCX-128 cartridge")
-
 //-------------------------------------------------
 //  mc10_pak_device - constructor
 //-------------------------------------------------
 
 mc10_pak_mcx128_device::mc10_pak_mcx128_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
-	:mc10_pak_mcx128_device(mconfig, MC10_PAK_MCX128, tag, owner, clock)
-{
-}
-
-mc10_pak_mcx128_device::mc10_pak_mcx128_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock)
-	: device_t(mconfig, type, tag, owner, clock)
+	: device_t(mconfig, MC10_PAK_MCX128, tag, owner, clock)
 	, device_mc10cart_interface(mconfig, *this)
 	, m_share(*this, "ext_ram", 1024*128, ENDIANNESS_BIG)
 	, m_view(*this, "mcx_view")
 	, m_bank(*this, "bank%u", 0U)
 {
+	const char *driver_name = mconfig.gamedrv().name;
+	if (strcmp(driver_name, "alice") == 0)
+	 	set_default_bios_tag("alice_mcx128");
+	else
+		set_default_bios_tag("mc10_mcx128");
```

Comment:
You can’t do this.  If you want to change the default BIOS in a particular system, you need to do it with `set_option_default_bios` on the slot when adding the slot to machine configuration.

See examples:
* https://github.com/mamedev/mame/blob/4183b96612eb588a76b271ca610604a607469fb7/src/devices/bus/abcbus/abc890.cpp#L78
* https://github.com/mamedev/mame/blob/4183b96612eb588a76b271ca610604a607469fb7/src/mame/pc/pc.cpp#L1115

Reply:
I've looked at it, and it worked. Then found the default_bios() on the slot_option returned by the add_option() function. I used this.